OpenAI/Packages/com.openai.unity/Runtime/Threads/ThreadExtensions.cs

```diff
@@ -192,11 +186,21 @@ public static async Task<RunResponse> CreateRunAsync(this ThreadResponse thread,
         /// Create a run.
         /// </summary>
         /// <param name="thread"><see cref="ThreadResponse"/>.</param>
-        /// <param name="assistantId">Id of the assistant to use for the run.</param>
+        /// <param name="assistant">The <see cref="AssistantResponse"/> to use for the run.</param>
         /// <param name="cancellationToken">Optional, <see cref="CancellationToken"/>.</param>
         /// <returns><see cref="RunResponse"/>.</returns>
-        public static async Task<RunResponse> CreateRunAsync(this ThreadResponse thread, string assistantId, CancellationToken cancellationToken = default)
-            => await thread.Client.ThreadsEndpoint.CreateRunAsync(thread, new CreateRunRequest(assistantId), cancellationToken);
+        public static async Task<RunResponse> CreateRunAsync(this ThreadResponse thread, AssistantResponse assistant, CancellationToken cancellationToken = default)
+        {
+            var request = new CreateRunRequest(
+                assistant,
+                model: assistant.Model,
+                instructions: assistant.Instructions,
+                tools: assistant.Tools,
+                temperature: assistant.Temperature,
+                topP: assistant.TopP,
+                responseFormat: assistant.ResponseFormat);
```

Comment:
@StephenHodgson super nice with all the upgrades! Working very well so far.
I noticed an odd behavior here though, where the Assistant instructions, tools etc. are re-sent at every Run.
It causes an error where the Tools ID cannot be found on repeated tool calls.

But it also seems to not be needed, since the Assitant info has already been sent to the server. 
I changed the above to simply pass the Assistant ID and it fixed the issue. 
```var request = new CreateRunRequest(assistant);```

Reply:
Ok, I'll check it out.

Reply:
I suppose one could always use this the first time but then use the CreateRunRequeset overload for subsequent requests.

I assumed people would only use this specific overload for the first run request of a thread.

Reply:
Ahh interesting. Can you clarify the sequence of method calls? 
I don't see `CreateRunRequest()` used in the sample documentation.

Reply:
I thought I used both in the documentation. 

I added each of the overloads as possibilities.

Reply:
All of the method extensions are optional